### PR TITLE
Remove pango.c (unused)

### PR DESCRIPTION
**Pull request by c-edw**
See review on #23.

### CODE REVIEW

File: pango.c [hunk 1]
```diff
@@ -93,15 +93,17 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	char *buf = malloc(max_chars);
+	va_list args;
+
+	uint32_t size = vsnprintf(NULL, 0, fmt, args);
```
Comment (emersion) [inline pango.c]: `vsnprintf` returns an `int`. Can you change the type of `size` to match?

File: pango.c [hunk 2]
```diff
@@ -93,15 +93,17 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
...
+	char *buf = malloc(size);
```
Comment (emersion) [inline pango.c]: Need to allocate one byte more, for the terminating NULL byte (see the manpage).

Comment (c-edw) [reply pango.c]: I see, `vsnprintf`s return value excludes the NULL byte.

Comment (emersion) [reply pango.c]: Yeah, that's not the best API ever. :S

File: pango.c [hunk 3]
```diff
@@ -93,15 +93,17 @@ PangoLayout *get_pango_layout(cairo_t *cairo, const char *font,
 
 void get_text_size(cairo_t *cairo, const char *font, int *width, int *height,
 		int *baseline, double scale, bool markup, const char *fmt, ...) {
-	char *buf = malloc(max_chars);
+	va_list args;
+
+	int size = vsnprintf(NULL, 0, fmt, args);
```
Comment (emersion) [inline pango.c]: `args` is undefined at this point, maybe `va_start` should be moved above?

Comment (emersion) [reply pango.c]: Disclaimer: I'm not a VA expert

Comment (c-edw) [reply pango.c]: Yep: ```
The va_start() macro initializes ap for subsequent use by va_arg() and va_end(), and must be called first.
```

Comment (c-edw) [reply pango.c]: Why is that? `vsnprintf` doesn't mutate `args`, does it?

Comment (emersion) [reply pango.c]: See `va_arg`'s manpage.

Comment (c-edw) [reply pango.c]: Actually, this file doesn't even seem to be in use. Is it an artifact of moving it to a separate repo or does it need to stay?

Comment (emersion) [reply pango.c]: Oh, if it's unused, please remove it.

Comment (c-edw) [reply pango.c]: Sure, I'll sweep for any others while I'm at it.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**